src/utils/csv.ts

```diff
@@ -0,0 +1,35 @@
+import { writeToString } from "@fast-csv/format";
+
+type RowInterface = {
+  [key: string]: string;
+};
+
+const getCSVRows = (rows: readonly RowInterface[]) => {
+  let results: RowInterface[] = [];
+  rows.map((row: RowInterface) => {
+    results.push(row);
+    return "";
+  });
+  return results;
+};
+
+export const downloadGeneratedCSVFile = async (
+  // Rows needed to be normal objects but as input type is WithId<User> I can not pass it to @fast-csv/format lib so I need to use any
+  // eslint-disable-next-line
+  rows: readonly any[],
+  headers: string[]
+) => {
+  const csvContent = await writeToString(getCSVRows(rows), {
+    headers: headers,
+    quote: true,
+  });
+  const encodedUri = encodeURI("data:text/csv;charset=utf-8," + csvContent);
+  const link = document.createElement("a");
+  link.setAttribute("href", encodedUri);
```

Comment:
Why are you manually constructing/encoding a URI here? I detailed how to do this already in https://github.com/sparkletown/internal-sparkle-issues/issues/897#issuecomment-877791352

Reply:
fixed in a2c3ebfeb0bd6fc1d4c317f56172510fc58adcab

Reply:
fixed in a2c3ebfeb0bd6fc1d4c317f56172510fc58adcab